transmogrifier/sources/xml/ead.py

```diff
@@ -17,72 +18,36 @@
 class Ead(XMLTransformer):
     """EAD transformer."""
 
-    def get_optional_fields(self, xml: Tag) -> dict | None:
+    def get_optional_fields(self, source_record: Tag) -> dict | None:
         """
         Retrieve optional TIMDEX fields from an EAD XML record.
 
         Overrides metaclass get_optional_fields() method.
 
         Args:
-            xml: A BeautifulSoup Tag representing a single EAD XML record.
+            source_record: A BeautifulSoup Tag representing a single EAD XML record.
         """
         fields: dict = {}
 
-        source_record_id = self.get_source_record_id(xml)
+        source_record_id = self.get_source_record_id(source_record)
 
-        if collection_description := xml.metadata.find("archdesc", level="collection"):
-            pass
-        else:
-            message = (
-                f"Record ID {self.get_source_record_id(xml)} is missing archdesc element"
-            )
-            logger.error(message)
-            return None
+        # <archdesc> and <did> elements are required when deriving optional fields
+        collection_description = self._get_collection_description(source_record)
+        collection_description_did = self._get_collection_description_did(source_record)
```

Comment:
Wondering if these calls are still necessary up top, it seems they would raise the `SkippedRecordEvent` whenever they are called by the field methods

Reply:
I left a comment that might be related: my assumption is that these will be removed once _fully_ refactored to field methods?

Reply:
Ah, didn't see your comments before my review but that makes sense as a plan!

Reply:
See [response](https://github.com/MITLibraries/transmogrifier/pull/185#discussion_r1621324261); PR description updated with note. Sorry for the confusion!